### PR TITLE
Replace `TextControl` with `SearchControl` in `ContentSearch` Component

### PR DESCRIPTION
**Pull request by psorensen**
### How to test the Change

1. Insert Content Picker or Post Picker block in /example 
2. Confirm search component is used and block is operating normally

### Changelog Entry

- Replaces TextControl with SearchControl in 'content-search' component
- Adds Content Picker block to test and demonstrate content-picker component
- Updates documentation to include new prop and updates demo gif

### Credits

Props @psorensen 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Review by Antonio-Laguna — APPROVED** [review]
This LGTM! Thanks @psorensen !

Will let @fabiankaegy chime in since he might more opinion here!

File: components/content-search/index.js [hunk 1]
```diff
@@ -343,12 +344,13 @@ const ContentSearch = ({
 	return (
 		<StyledComponentContext cacheKey="tenup-component-content-search">
 			<NavigableMenu onNavigate={handleOnNavigate} orientation="vertical">
-				<TextControl
-					label={label}
+				<SearchControl
```
Comment (fabiankaegy) [inline components/content-search/index.js:347]:
```suggestion
				<SearchControl
					__next40pxDefaultSize
```

In order to look closer to all the core component let also use the future default style now already :)

File: components/content-search/index.js [hunk 2]
```diff
@@ -357,6 +359,8 @@ const ContentSearch = ({
 					onBlur={() => {
 						setIsFocused(false);
 					}}
+					size="compact"
```
Comment (fabiankaegy) [inline components/content-search/index.js]: Actually I think for us here the not compact size is better

File: components/content-search/index.js [hunk 3]
```diff
@@ -357,6 +359,8 @@ const ContentSearch = ({
...
+					style={{ border: '1px solid rgb(30, 30, 30)' }}
```
Comment (fabiankaegy) [inline components/content-search/index.js]: I'm not sure we need this manual border. With the greater space it has this should look better without I think

**Review by fabiankaegy — CHANGES_REQUESTED** [review]
Thanks for making this update :) 

I've left some suggestions for what I personally think looks better and feels more native in the editor.

| Before | After |
|--------|------|
| CleanShot 2024-01-18 at 11 44 54@2x | CleanShot 2024-01-18 at 11 44 26@2x |

File: components/content-search/index.js (hunk 2 above)
Comment (psorensen) [reply components/content-search/index.js]: Removed

File: components/content-search/index.js (hunk 3 above)
Comment (psorensen) [reply components/content-search/index.js]: My concern here is that I cannot see the border of the input at all - maybe it's my eyes, but I don't even realize it's a search input without the border

Comment (psorensen) [reply components/content-search/index.js]: Ah - there was something with my monitor - probably a high contrast setting (ironic enough) that was preventing me from seeing the gray background of the input. I've removed the manual style.

**Review by fabiankaegy — APPROVED** [review]

**Pull request merged**